ADR/0033-product-version-controller.md

```diff
@@ -0,0 +1,493 @@
+  componentName: "cool-comp2-main"
+  source:
+    git:
+      context: ./
+      dockerfileUrl: Dockerfile
+      revision: fixed-rev
+      uri: git@github.com:example/comp2.git
+```
+
+Since the *ProjectDevelopmentStream* resource has a *toTemplate* property,
+following is the *ProjectDevelopmentStreamTemplate* that would be generated
+from it (Some extra whitespace was added to the YAML for readability):
+
+```
+apiVersion: pvc.konflux.dev/v1alpha1
+kind: ProjectDevelopmentStreamTemplate
+metadata:
+  name: my-cool-project-stream-template
+  ownerReference:
+    apiVersion: pvc.konflux.dev/v1alpha1
+    kind: Project
+    name: my-cool-project
+spec:
+  project: my-cool-project
+  variables:
+  - name: version
+    description: A version number for the new development stream
+
+  - name: cool-comp1-context
+    defaultValue: ./
+    description: Context directory for cool-comp1 component
+  - name: cool-comp1-dockerfileUrl
+    defaultValue: Dockerfile
+    description: Dockerfile location for cool-comp1 component
+  - name: cool-comp1-revision
+    defaultValue: {version}
+    description: Git revision for cool-comp1 component
+
+  - name: cool-comp2-context
+    defaultValue: ./
+    description: Context directory for cool-comp2 component
+  - name: cool-comp2-dockerfileUrl
+    defaultValue: Dockerfile
+    description: Dockerfile location for cool-comp2 component
+  - name: cool-comp2-revision
+    defaultValue: fixed-rev
+    description: Git revision for cool-comp2 component
+
+  resources:
+  - apiVersion: appstudio.redhat.com/v1alpha1
+    kind: Application
+    metadata:
+      name: "cool-app-{version}"
+      annotations:
+        pvc.konflux.dev/cloned-from: cool-app1-main
+    spec:
+      displayName: "Cool App {version}"
+
+  - apiVersion: appstudio.redhat.com/v1alpha1
+    kind: Component
+    metadata:
+      name: "cool-comp1-{version}"
+      annotations:
+        pvc.konflux.dev/cloned-from: cool-comp1-main
+    spec:
+      application: "cool-app-{version}"
+      componentName: "cool-comp1-{version}"
+      source:
+        git:
+          context: {cool-comp1-context}
+          dockerfileUrl: {cool-comp1-dockerfileUrl}
+          revision: {cool-comp1-revision}
+          uri: git@github.com:example/comp1.git
+
+  - apiVersion: appstudio.redhat.com/v1alpha1
+    kind: Component
+    metadata:
+      name: "cool-comp2-{version}"
+      annotations:
+        pvc.konflux.dev/cloned-from: cool-comp2-main
+    spec:
+      application: "cool-app-{version}"
+      componentName: "cool-comp2-{version}"
+      source:
+        git:
+          context: {cool-comp2-context}
+          dockerfileUrl: {cool-comp2-dockerfileUrl}
+          revision: {cool-comp2-revision}
+          uri: git@github.com:example/comp2.git
+```
+
+Following is a new *ProjectDevelopmentStream* resource making use of the
+template above:
+
+```
+apiVersion: pvc.konflux.dev/v1alpha1
+kind: ProjectDevelopmentStream
+metadata:
```

Comment:
Having a `ProductDevelopmentStream` resource which is able to name a product stream in its metadata could be really useful for augmenting SBOMs with the data needed for product security.